File: generate-activity-index.js
const fs = require("fs");
const path = require("path");

const repoRoot = __dirname;
const activityDataRoot = path.join(repoRoot, "data", "activity_data");
const activityCoreDir = path.join(activityDataRoot, "activity_core");
const outputFile = path.join(activityDataRoot, "activity_index.json");

function walk(dir, baseDir) {
  let results = [];

  const entries = fs.readdirSync(dir, { withFileTypes: true });

  for (const entry of entries) {
    const fullPath = path.join(dir, entry.name);

    if (entry.isDirectory()) {
      results = results.concat(walk(fullPath, baseDir));
    } else if (entry.isFile() && entry.name.endsWith(".json")) {
      const relative = path.relative(baseDir, fullPath).replace(/\\/g, "/");
      results.push(relative);
    }
  }

  return results;
}

if (!fs.existsSync(activityCoreDir)) {
  console.error("Could not find activity_core directory:");
  console.error(activityCoreDir);
  process.exit(1);
}

const files = walk(activityCoreDir, activityDataRoot).sort();

fs.writeFileSync(outputFile, JSON.stringify(files, null, 2));
console.log(`Wrote ${files.length} entries to ${outputFile}`);

File: generate-activity-index.py
from pathlib import Path
import json

# Repository root (where this script lives)
repo_root = Path(__file__).resolve().parent

# Target directories
activity_data_root = repo_root / "data" / "activity_data"
activity_core_dir = activity_data_root / "activity_core"

# Output file
output_file = activity_data_root / "activity_index.json"

if not activity_core_dir.exists():
    raise FileNotFoundError(f"Could not find activity_core directory: {activity_core_dir}")

# Recursively find all JSON files inside activity_core
files = sorted(
    str(path.relative_to(activity_data_root)).replace("\\", "/")
    for path in activity_core_dir.rglob("*.json")
)

# Write the index file
with open(output_file, "w", encoding="utf-8") as f:
    json.dump(files, f, indent=2)

print(f"Generated activity_index.json with {len(files)} entries.")
print(f"Location: {output_file}")